### PR TITLE
Updates the Roadmap for Q2/Q3

### PR DESCRIPTION
**Pull request by ellis-bigelow**
**What this PR does / why we need it**:
Proposed ROADMAP from KFServing Working Group Meetings.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Included in scope of #170

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/173)

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -1,7 +1,45 @@
 # KF Serving 2019 Roadmap
```
Comment (ellis-bigelow) [inline ROADMAP.md]: @ellis-bigelow , reorder these:

now 
-
-
-
future

File: ROADMAP.md [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
+    * Prepare KFServing to release v0.2 and v0.3 alongside Kubeflow 0.7.
+    * Integrate with `kfctl generate` and `kfctl apply`.
+    * Deploy as a [Kubernetes Application](https://github.com/kubernetes-sigs/application).
+    * Integrate with PVCs to enable integration with Kubeflow Pipelines
+    * Integrate with Fairing to enable deployment from a Notebook
+
+Objective: "Empower users to deeply understand their predictions and validate KFServing's static graph architecture."
+* Explainability
+    * Deploy a predictor and explainer, powered by Alibi.
+    * Deploy a predictor and explainer, powered by user specified explainer container.
```
Comment (jpe316) [inline ROADMAP.md]: Are the predictor and explainer deployed in parallel? Is there any preprocess / postprocess transform logic in place here?

File: ROADMAP.md [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
 
-### Core CUJs
+Objective: "Increase coverage of ML frameworks to support previously unsupported customer workloads."
+* Frameworks
```
Comment (jpe316) [inline ROADMAP.md]: We also need to support scikit-learn IMO. It is also fairly easy to support with existing scaffolding

File: ROADMAP.md [hunk 4]
```diff
@@ -1,7 +1,45 @@
 # KF Serving 2019 Roadmap
-## Q2 2019
+## Q3 2019
+### v0.3 Performance and Stability (ETA: September 1, 2019)
+Objective: "Prevent feature regressions with 80% e2e test coverage against a live Cluster."
+* Automated End-to-End Tests
+    * Execute against a Kubeflow maintained GKE Cluster.
```
Comment (jpe316) [inline ROADMAP.md]: Should also cover AKS

File: ROADMAP.md [hunk 5]
```diff
@@ -1,7 +1,45 @@
...
+    * Achieve >80% Test Coverage of Supported Features.
```
Comment (jpe316) [inline ROADMAP.md]: Hard to achieve when the full feature list is not specified. Are you talking about code coverage of the code specific to this implementation?

File: ROADMAP.md [hunk 6]
```diff
@@ -1,7 +1,45 @@
...
+
+Objective: "Prevent performance regressions across a known set of representative models."
+* Automated Performance Tests 
+    * Define a set of Models to test covering a wide array of usecases and frameworks.
```
Comment (jpe316) [inline ROADMAP.md]: are we testing predictive accuracy or latency / CPU / memory load?
@gogowings fyi

File: ROADMAP.md (hunk 2 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Yeah -- they're in parallel to enable the predictor to scale past the explainer. We've discussed pre/post processing extensively, but I need to add this to the roadmap. Likely unplanned for now.

File: ROADMAP.md (hunk 3 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Already supported

File: ROADMAP.md (hunk 4 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Agreed. Is this something Azure would consider contributing?

File: ROADMAP.md (hunk 5 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Agreed. 80% is a proxy for "good but not perfect" coverage. Features are often difficult to count.

File: ROADMAP.md (hunk 4 above)
Comment (jpe316) [reply ROADMAP.md]: 100% - defer to @rakelkar but we pretty much have to do this

File: ROADMAP.md (hunk 2 above)
Comment (jpe316) [reply ROADMAP.md]: got it - i am wondering if customers will have the ability to include any custom code / functions alongside their model data files. would this be expressed in a model class file?

File: ROADMAP.md (hunk 6 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Infrastructure. These are on a known set of pretrained models.

File: ROADMAP.md [hunk 7]
```diff
@@ -23,20 +61,16 @@ Objective: "Reduce the total cost of ownership for models by minimizing the delt
...
+    * GPU Sharing.
+* Flexible Inference Graphs [MLGraph CRD](https://github.com/SeldonIO/mlgraph).
+    * Model Experimentation.
+    * Ensembling.
+    * Multi Arm Bandit.
+* Skew and Outlier Detection.
+    * Online support in graph.
+    * Offline support with Payload Logging.
+* Meta-Protocol Definition.
+    * Unify disparate protocols across frameworks.
```
Comment (animeshsingh) [inline ROADMAP.md]: Add 'Bias Detection' as one of the future work here

Comment (rakelkar) [reply ROADMAP.md]: Also mini-batch/adaptive batching support

File: ROADMAP.md [hunk 8]
```diff
@@ -1,7 +1,64 @@
 # KF Serving 2019 Roadmap
-## Q2 2019
+## Q3 2019
+### v0.2 Integrate with the ML Ecosystem (ETA: August 15, 2019)
```
Comment (animeshsingh) [inline ROADMAP.md]: And have examples for 'Cloud Providers':  e.g. getting it running on Azure, IBM Cloud etc

File: ROADMAP.md [hunk 9]
```diff
@@ -1,7 +1,64 @@
...
+Objective: "Continue to simplify the user experience by deeply integrating with the Kubeflow Ecosystem."
+* Kubeflow Integration
+    * Prepare KFServing to release v0.2 and v0.3 alongside Kubeflow 0.7.
+    * Integrate with `kfctl generate` and `kfctl apply`.
+    * Deploy as a [Kubernetes Application](https://github.com/kubernetes-sigs/application).
+    * Integrate with PVCs to enable integration with Kubeflow Pipelines
```
Comment (animeshsingh) [inline ROADMAP.md]: PVCs support should be independent of Pipelines

File: ROADMAP.md (hunk 8 above)
Comment (ellis-bigelow) [reply ROADMAP.md]: Sure

**Comment by jinchihe** [comment]
@ellis-bigelow Shall we add the KFServing Python SDK in the Q3 2019 v0.2 roadmap scope? Thanks.

/cc @yuzisun

**Comment by ellis-bigelow** [comment]
> @ellis-bigelow Shall we add the KFServing Python SDK in the Q3 2019 v0.2 roadmap scope? Thanks.
> 
> /cc @yuzisun

We don't need to include everything on the roadmap. I think of it as more of a nice to have than a core v0.2 deliverable. Does that sound okay?
